src/EventRegistry.cc

```diff
@@ -164,4 +165,45 @@ void EventGroup::Disable() {
 
 void EventGroup::AddFunc(detail::ScriptFuncPtr f) { funcs.insert(f); }
 
+bool EventRegistry::RegisterMetadata(EnumValPtr id, TypePtr type) {
+    static const auto& metadata_id_type = id::find_type<EnumType>("EventMetadata::ID");
+
+    if ( metadata_id_type != id->GetType() )
+        return false;
+
+    auto id_int = id->Get();
+    if ( id_int < 0 ) {
+        zeek::reporter->InternalError("Negative enum value %s: %" PRId64, obj_desc_short(id.get()).c_str(), id_int);
+        return false; // unreached
+    }
+
+    zeek_uint_t id_uint = static_cast<zeek_uint_t>(id_int);
+
+    auto it = event_metadata_types.find(id_uint);
+
+    // If type contains anything with any, like table[count] of any, we should reject it.
+    //
+    //     cb = AnyTypeChecker()
+    //     type->Traverse(cb)
+    if ( it != event_metadata_types.end() )
+        return same_type(it->second.Type(), type);
```

Comment:
We can avoid putting `it` into the function scope.

I was briefly also wondering about logging of mismatched types here, but we seem to do that elsewhere on another layer already.

```suggestion
    if ( auto it = event_metadata_types.find(id_uint); it != event_metadata_types.end() )

        return same_type(it->second.Type(), type);
```